content/articles/2019-10-discrim-0-01.Rmarkdown

```diff
@@ -0,0 +1,217 @@
+---
+title: discrim 0.0.1
+date: 2019-10-16
+slug: discrim-0-0-1
+author: Max Kuhn
+categories: [package]
+description: >
+    The first version of discrim (0.0.1) is on CRAN. 
+photo:
+  url: https://unsplash.com/photos/4op9_2Bt2Eg
+  author: Teo Duldulao
+---
+
+```{r setup, include = FALSE}
+knitr::opts_chunk$set(
+  collapse = TRUE, comment = "#>", 
+  fig.width = 7, 
+  fig.align = 'center',
+  fig.asp = 0.618, # 1 / phi
+  out.width = "700px"
+)
+options(digits = 3)
+library(earth)
+library(klaR)
+library(ggplot2)
+theme_set(theme_bw())
+```
+
+The new package `discrim` contains `parsnip` bindings for additional classification models, including:
+
+ * Linear discriminant analysis (LDA, simple and L2 regularized)
+ * Regularized discriminant analysis (RDA, via [Friedman (1989)](https://scholar.google.com/scholar?hl=en&as_sdt=0%2C7&q=%22Regularized+Discriminant+Analysis%22&btnG=))
+ * [Flexible discriminant analysis](https://scholar.google.com/scholar?hl=en&as_sdt=0%2C7&q=%22Flexible+discriminant+analysis%22&btnG=) (FDA) using [MARS features](https://scholar.google.com/scholar?hl=en&as_sdt=0%2C7&q=%22multivariate+adaptive+regression+splines%22&btnG=)
+ * Naive Bayes models 
+
+The package can also be used as a template for adding new models to `tidymodels` without having to directly involve `parsnip`. 
+
+As an example, the package contains a simulated data set with two factors and two classes: 
+
+```{r startup}
+library(tidyverse)
+library(rsample)
+library(discrim)
+library(earth)
+
+data("parabolic", package = "rsample")
+parabolic
+
+ggplot(parabolic, aes(x = X1, y = X2)) + 
+  geom_point(aes(col = class), alpha = .3) + 
+  coord_equal() + 
+  theme(legend.position = "top")
+```
+
+How would a flexible discriminant model do here? We'll split the data then fit the model:
+
+```{r fda-1}
+set.seed(115)
+data_split <- initial_split(parabolic, prop = 2/3)
+data_tr <- training(data_split)
+data_te <- testing(data_split)
+
+fda_mod <- discrim_flexible() %>% set_engine("earth")
+
+fda_fit <- 
+  fda_mod %>% 
+  fit(class ~ X1 + X2, data = data_tr)
+
+fda_fit 
+```
+
+Since no model tuning parameters were specified, the MARS algorithm follows its own internal method for optimizing the number of features that are included in the model. The underlying MARS model is
```

Comment:
```suggestion
Since no model tuning parameters were specified, the MARS algorithm follows its own internal method for optimizing the number of features that are included in the model. The underlying MARS model is:
```